src/functions/read.ts

```diff
@@ -27,16 +28,23 @@ interface createReadOptions {
    * @default `minimal`
    */
   createMode?: RCGenerationMode | RCGenerationMode[];
+
+  /**
+   * RC automatic caching option. This option allows to cache a configuration passed in parameter.
+   *
+   * @default false
+   */
+  isMemoize?: boolean ;
 }
 
-export type readOptions = RequireAtLeastOne<createReadOptions, "createIfDoesNotExist" | "createMode">;
+export type readOptions = RequireAtLeastOne<createReadOptions, "createIfDoesNotExist" | "createMode" | "isMemoize">;
 
 export async function read(
   location = process.cwd(),
   options: readOptions = Object.create(null)
 ): Promise<Result<RC, NodeJS.ErrnoException>> {
   try {
-    const { createIfDoesNotExist = Boolean(options.createMode), createMode } = options;
+    const { createIfDoesNotExist = Boolean(options.createMode), createMode, isMemoize = false } = options;
```

Comment:
```js
const { createIfDoesNotExist = Boolean(options.createMode), createMode, memoize: memoizeRc = false } = options;
```

An idea to avoid collision